src/GovernorCountingFractional.sol

```diff
@@ -98,13 +98,17 @@ abstract contract GovernorCountingFractional is Governor {
         uint256 weight,
         bytes memory voteData
     ) internal virtual override {
-        require(!_proposalVotersHasVoted[proposalId][account], "GovernorCountingFractional: vote already cast");
-        _proposalVotersHasVoted[proposalId][account] = true;
+        require(weight > 0, "GovernorCountingFractional: no weight");
+        if (_proposalVotersWeightCast[proposalId][account] == weight) {
```

Comment:
Might be better to defensively check `>=`

Reply:
Thanks. Done in b202e74